service/controller/resource/nodes/create.go

```diff
@@ -0,0 +1,20 @@
+package nodes
+
+import (
+	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+)
+
+// IsClusterCreating check if the cluster is being created.
+// TODO: move to service
+func IsClusterCreating(cr providerv1alpha1.AzureConfig) bool {
```

Comment:
Maybe this one could live in the `key` package?

Reply:
It can be moved outside of resource definitely, also `func AnyNodesOutOfDate(ctx context.Context) (bool, error)`.

Now when you say it `key` seems to be a nice option. I was also thinking about maybe some package/service under `service/controller/internal`. @tuommaki any thoughts here?

Reply:
We use the `key` package for simple getter functions. Not sure if `InClusterCreating` has too much business logic, I think not, but `AnyNodesOutOfDate` has too much business logic to be in the `key` package, I believe.

Reply:
Moved `IsClusterCreating()` to `key` package.